guidelines/communicate.md

```diff
@@ -0,0 +1,62 @@
+# Ways to communicate
+
+The purpose of this page is to provide an overview of the three different ways to interact with the API and suggest when you might use each particular one.
+The main way to integrate with Mews is to use the API [Operations](../operations/README.md) to make requests, e.g. to fetch data or to request some other action.
+However, we also support [Webhooks](../webhooks/README.md) and [WebSockets](../websockets/README.md) for special cases where polling is not efficient or not real-time enough.
+
+> ### Three ways to communicate
+>
+> * [API Operations](#API-operations)
+> * [Webhooks](#webhooks)
+> * [WebSockets](#websockets)
+
+## API Operations
+
+Most of the functionality in the API is in the form of API Operations.
+These operations work on a stateless, request-response basis. That means when you want to interact, you choose the moment and send a request, to which you will receive a response.
+If the request is for information, then the response will normally contain the requested information.
+If the request is to make an action of some sort, then the response may just be a success message to tell you the action was performed by Mews ok.
+
+* For the details of individual API operations, see [Operations](../operations/README.md).
+
+But what if you want to know when some event happens in the Mews system?
+Using the standard request-response model, you would need to keep asking "has this event happened yet?".
+This is called polling, and is not very efficient. And that's where Webhooks and WebSockets come in.
+
+## Webhooks
+
+A Webhook is the equivalent of Mews saying "don't keep asking if this event has happened yet, instead we'll let you know when it's happened. Give us your contact details and we'll call you."
+The contact details in this case is an accessible URL, provided by you.
+
+![](../.gitbook/assets/webhooks-sequence.png)
+
+In summary, API Operations are hosted on the Mews side of the integration and they follow the on-demand request model, whereas Webhooks are hosted on the partner side of the integration, and they follow the event model.
+Of course a Webhook is still a web API using request-response because that is how HTTP works, but in this case Mews is doing the requesting; and rather than making the request to get information, Mews is making the request to provide you with information based on some event that occurred in the Mews system.
+
+* For the details of individual Webhooks, the events that generate them and the structure of the data that's sent, see [Webhooks](../webhooks/README.md).
+
+### When are Webhooks generated?
+
+A Webhook is generated when some event occurs in the system, however they are 'near real-time' and not 'true real-time'.
+Instead events are queued when they occur, with the queue being periodically read and the Webhooks generated.
```

Comment:
@vitasamek I don't think we should be exposing such technical details, it makes us look bad, because the fact that we're not able to send webhooks in real time is pretty basic flaw. So it's also big opportunity for technical roadmap, having such delays in webhooks isn't great.

Reply:
I could replace the text with this: "A Webhook is generated when some event occurs in the system, however note these are not true real-time events, there may be a lag of up to several minutes (depending on circumstances and on system parameters) between the event occurring and the Webhook being generated. For most use cases that is absolutely fine, but there are some use cases that require true real-time event information, e.g. generating digital keys. For this case, we have WebSockets." That removes the technical details and is not specific about the duration of the lag, but makes it clear there is a lag. We offer an alternative with WebSockets so I don't think it makes us look bad, it depends on the partner's use case.

Reply:
The websockets are another topic - I'm not sure we should even disclose the fact that we have them. Especially for server-to-server integrations, which should be the majority, they're poor from perspective of developer experience of the integration partner. So that could be part of the initiative I mentioned above - making webhooks real-time, including the websocket events, getting rid of websockets publicly and using them only for kiosks.

Reply:
@siroky All those topics were covered. We discussed the disadvantages of WebSockets ... If it's only about exposing implementation let's remove it.

Reply:
@MikeAdamsMews the fact that we have websockets currently doesn't mean we have to expose it. E.g. if we no longer want for partners to adopt it, we shouldn't expose it.

Reply:
For additional context, some partners still need WebSockets - e.g. some digital guest app or mobile keys - because it's real time. For example, the app will only create/authorise a key when a reservation is checked-in, and the guest might already be on their way to their room, and a 2-minute delay there might detract from a smooth guest experience. 

So far, we haven't seen other use cases where it's critical to have real-time notifications.

re @MikeAdamsMews 's last comment, knowing the partners, those interested in Webhooks will likely still write partnersuccess@ to ask how many minutes of delay exactly they can expect (if we say "several minutes" in the API docs)